### PR TITLE
RHCLOUD - 31941 Workspaces Overview - About tab

### PR DESCRIPTION
**Pull request by aferd**
### Description

Workspaces overview page - About access management tab.

This PR covers the information shown under the "About access management" tab. It **does not** include the tab itself or any work under the "Dashboard" tab.

[RHCLOUD-31941](https://issues.redhat.com/browse/RHCLOUD-31941)

---

#### After:
Screenshot 2024-08-09 at 11 38 00 AM

---

### Checklist ☑️
- [ ] PR only fixes one issue or story 
- [ ] Change reviewed for extraneous code 
- [ ] UI best practices adhered to 
- [ ] Commits squashed and meaningfully named 
- [ ] All PR checks pass locally (build, lint, test, E2E)

##
- [ ] _(Optional) QE: Needs QE attention (OUIA changed, perceived impact to tests, no test coverage)_
- [ ] _(Optional) QE: Has been mentioned_
- [ ] _(Optional) UX: Needs UX attention (end user UX modified, missing designs)_
- [ ] _(Optional) UX: Has been mentioned_
##

### CODE REVIEW

File: src/smart-components/workspaces/overview/about-access-tab.tsx [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+        >
+          {/* to do - add migration visualization when ready */}
+          <Card ouiaId="animation-card">
+            <CardTitle>A cool animation</CardTitle>
+            <CardBody>will go here</CardBody>
+            <CardFooter>when its ready</CardFooter>
+          </Card>
+        </ExpandableSection>
+
+        <br></br>
```
Comment (fhlavac) [inline src/smart-components/workspaces/overview/about-access-tab.tsx:74]:
```suggestion
        <br/>
```
multiple places below

**Comment by fhlavac** [comment]
@aferd can you please make the strings localized?

File: src/smart-components/workspaces/overview/custom-data-list-item.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  icon,
+  heading,
+  linkTitle,
+  linkTarget,
+  expandableContent,
+  isRedirect,
+  isExpanded,
+}) => {
+  const navigate = useNavigate();
+  const iconElement: React.ReactNode | null = IconMapper[icon] || null;
```
Comment (fhlavac) [inline src/smart-components/workspaces/overview/custom-data-list-item.tsx:50]:
```suggestion
  const iconElement: React.ReactNode = IconMapper[icon] || null;
```

File: src/smart-components/workspaces/overview/custom-data-list-item.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+          {linkTitle && linkTarget && (
+            <DataListAction aria-labelledby="item1 action1" id="action1" aria-label="Actions" isPlainButtonAction>
+              <Button
+                component="a"
+                href={linkTarget}
+                variant="link"
+                onClick={(e) => {
+                  if (!isRedirect) {
+                    e.preventDefault();
+                    navigate(linkTarget.replace('/preview', ''));
```
Comment (fhlavac) [inline src/smart-components/workspaces/overview/custom-data-list-item.tsx:88]: Is this still needed?

File: src/smart-components/workspaces/overview/custom-data-list-item.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                  }
+                }}
+              >
+                {linkTitle}
+              </Button>
+            </DataListAction>
+          )}
+        </DataListItemRow>
+        <DataListContent aria-label={heading + ' - Detailed Explanation'} id="expand1" isHidden={!expanded}>
+          <p>{expandableContent}</p>
```
Comment (fhlavac) [inline src/smart-components/workspaces/overview/custom-data-list-item.tsx:98]: We may want to use PF Text with `p` variant

File: src/smart-components/workspaces/overview/about-access-tab.tsx [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+        <Title headingLevel="h2" className="pf-u-mb-md" data-ouia-component-id="header-title">
+          Get started with workspaces
+        </Title>
+        <Text component={TextVariants.p}>
+          Workspaces let&apos;s you group related assets together (such as RHEL hosts). This simplifies management and user access control.
+        </Text>
+        <br></br>
+        <ExpandableSection
+          toggleText="Show me how my assets and permissions will be organized into workspaces"
+          onToggle={onToggle}
```
Comment (fhlavac) [inline src/smart-components/workspaces/overview/about-access-tab.tsx:63]:
```suggestion
          onToggle={(_e: React.MouseEvent, isExpanded: boolean) => 
    setIsExpanded(isExpanded)}
```

File: src/smart-components/workspaces/overview/about-access-tab.tsx [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+
+const WorkspacesOverview = () => {
+  const workspacesIcon = '/apps/frontend-assets/rbac-landing/rbac-landing-icon.svg';
+  const bindingsIcon = '/apps/frontend-assets/rbac-landing/workspaces-bindings-icon.svg';
+  const rolesIcon = '/apps/frontend-assets/rbac-landing/workspaces-roles-icon.svg';
+
+  const [isExpanded, setIsExpanded] = React.useState(false);
+  const onToggle = (_event: React.MouseEvent, isExpanded: boolean) => {
+    setIsExpanded(isExpanded);
+  };
```
Comment (fhlavac) [inline src/smart-components/workspaces/overview/about-access-tab.tsx:38]:
```suggestion
```

File: src/smart-components/workspaces/overview/about-access-tab.tsx [hunk 7]
```diff
@@ -0,0 +1,254 @@
...
+                </Button>
+              }
+              ouiaId="groups-service-card"
+            />
+          </GalleryItem>
+          <GalleryItem>
+            <ServiceCard
+              isFullHeight
+              title="Role"
+              subtitle=""
```
Comment (fhlavac) [inline src/smart-components/workspaces/overview/about-access-tab.tsx:112]:
```suggestion
```

File: src/smart-components/workspaces/overview/about-access-tab.tsx [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+                </Button>
+              }
+              ouiaId="role-service-card"
+            />
+          </GalleryItem>
+          <GalleryItem>
+            <ServiceCard
+              isFullHeight
+              title="Bindings"
+              subtitle=""
```
Comment (fhlavac) [inline src/smart-components/workspaces/overview/about-access-tab.tsx:127]:
```suggestion
```

File: src/smart-components/workspaces/overview/custom-data-list-item.tsx [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+
+export enum IconName {
+  KEY = 'access',
+  USERS = 'users',
+}
+
+interface CustomDataListItemProps {
+  icon: IconName;
```
Comment (fhlavac) [inline src/smart-components/workspaces/overview/custom-data-list-item.tsx:26]: Maybe we could just pass the icon itself and avoid the enum

File: src/smart-components/workspaces/overview/about-access-tab.tsx [hunk 10]
```diff
@@ -0,0 +1,254 @@
...
+import { Table, Tbody, Tr, Td } from '@patternfly/react-table';
+
+const VIEW_DEFAULT_GROUPS = 'https://console.redhat.com/iam/user-access/groups';
+// to do - update link when available
+const GRANT_ACCESS = '';
+
+const WorkspacesOverview = () => {
+  const workspacesIcon = '/apps/frontend-assets/rbac-landing/rbac-landing-icon.svg';
+  const bindingsIcon = '/apps/frontend-assets/rbac-landing/workspaces-bindings-icon.svg';
+  const rolesIcon = '/apps/frontend-assets/rbac-landing/workspaces-roles-icon.svg';
```
Comment (fhlavac) [inline src/smart-components/workspaces/overview/about-access-tab.tsx:33]: It is generally a better practice to move these constants above the component so they are not recreated on every render

File: src/smart-components/workspaces/overview/about-access-tab.tsx (hunk 7 above)
Comment (aferd) [reply src/smart-components/workspaces/overview/about-access-tab.tsx:112]: it errors when I try and remove subtitle, that's why I added it as an empty string
`Property 'subtitle' is missing in type '{ isFullHeight: true; title: string; description: string; icon: Element; footer: Element; ouiaId: string; }' but required in type 'ServiceCardProps'.ts(2741)`

Comment (InsaneZein) [reply src/smart-components/workspaces/overview/about-access-tab.tsx:112]: Oh yeah subtitle isn't an optional prop. I could open up a PR to make it so if you want?

Comment (aferd) [reply src/smart-components/workspaces/overview/about-access-tab.tsx:112]: Could be worth it? I know this is all for this specific use case, but if UX is designing in this way, might make sense.

Comment (InsaneZein) [reply src/smart-components/workspaces/overview/about-access-tab.tsx:107]: went ahead and made a pr https://github.com/patternfly/react-component-groups/pull/261

**Pull request merged**